### DISCUSSION TITLE
For production environments, Whether to host the PostgreSQL metadata database separately or as a Kubernetes pod. What is recommended ?

### DISCUSSION THREAD

**Discussion by michaelkovatt** [apache/superset#29383] [Q&A / Help] [answered]
Hi Team,

As of now superset ships Postgres database as part of Kubernetes setup as a separate Kubernetes POD that hosts the Postgres database to store all metadata related information (charts, dashboards, roles etc.). Currently, when we configure Apache superset there is a dedicated POD exclusively hosting Postgres.

In a production setup,  _**Is it better to host the Postgres metadata database (Managed setup) separately ?. Or is it fine to keep it as a Kubernetes POD itself**_ ?

_**What is the best practice and recommendation from superset team ?**_

_Current setup, where we have a seperate postgres sql pod hosting postgres db. Example below:_
![image](https://github.com/apache/superset/assets/151566993/ac6b5b6e-5699-4900-9dba-c12f4d646fcb)

**Comment by nytai** [comment, marked as answer]
the k8s pod is only included as a convenience feature to allow you to bring up superset with a k8s cluster and a few commands. Using a dedicated database service for postgres is generally recommended

**Reply by michaelkovatt** [reply]
thank you @nytai.
